src/v/cluster/rm_stm.cc

```diff
@@ -451,6 +451,14 @@ ss::future<tx_errc> rm_stm::do_prepare_tx(
   model::producer_identity pid,
   model::tx_seq tx_seq,
   model::timeout_clock::duration timeout) {
+    if (is_transaction_ga()) {
```

Comment:
we need to support prepare for `old'` txn coordinator and `new` partitions

Reply:
You mean do not put this check inside?

Reply:
right, `new` still may work with `old'` which still sends `prepare`